Add FPS and Resolution Settings

Elliesaur opened the pull request:
Allows selecting 1920x1080 or 2560x1440, with FPS selection as 60, and 120.
These are added as config options and elements also added to the overlay.

240 FPS technically can work, but I guess this is limited to AV1 encoding/Windows-specific implementation which I have not looked into.

Source for changes: https://gist.github.com/a9udn9u/85e0c8e863db85c5b62320766bd7b2e9#file-pac-script-js


## Screenshots
2560x1440 at 120 FPS (my screens are 1080p):
<img width="1919" height="1080" alt="image" src="https://github.com/user-attachments/assets/8a29c0ec-3459-4b85-8cab-8572f0a28cfa" />


1920x1080 at 120 FPS (my screens are 1080p):
<img width="1919" height="1079" alt="image" src="https://github.com/user-attachments/assets/9f807a2f-0306-4ebc-b2a6-30193c88ceee" />

AstralVixen reviewed (changes requested):
Please also add the new config values to `src/utils/cloudSync.ts`

Add to `const configUpdate` as:
`monitorWidth: data.gfimonwidth ?? 1920,`
`monitorHeight: data.gfimonheight ?? 1080`
`framesPerSecond: data.gfifps ?? 60, `

Add to `const userData` as:
`gfimonwidth: config.monitorWidth,`
`gfimonheight: config.monitorHeight,`
`gfifps: config.framesPerSecond, `

tOmsk is currently on vacation so your PR will take a while to be approved, since it requires both of our approvals to be merged. And also I'm unable to test if the FPS and Resolution settings do actually work, as I don't have a performance or ultimate plan so tOmsk will have to test it. However if you could drop some screenshots of it working that would be great!

Elliesaur commented:
Thanks for the quick review, I have updated the source to suit the requirements for cloud sync.

I have also added two screenshots. I was not able to get 240 FPS working (perhaps this is something Windows-specific?). I have removed 240 FPS as a selection from the dropdown.

All good with the wait, I have Ultimate so I could test.

AstralVixen reviewed (dismissed):
Thanks for making the changes and providing the screenshots. And overall for this whole feature. We didn't even know where to start, so you saved us a ton of time.

I'll give an approval review now, we will have to wait until tOmsk also reviews it.

MrBretze commented:
I have tried your PR in my system (CPU AMD (R9 7950X3D ) and GPU AMD (RX 6900XT) on Bazzite OS but it doesn't work correctly at 1440P with the ultimate subscription.
I have found [this one on Reddit,](https://www.reddit.com/r/LinuxOnAlly/comments/1kslnd1/get_better_decode_time_in_xcloud_on_bazzite/) and i added the argument `--enable-features=AcceleratedVideoDecodeLinuxZeroCopyGL,AcceleratedVideoDecodeLinuxGL,VaapiIgnoreDriverChecks` to the [/src/electron/main.ts](https://github.com/Elliesaur/GeForce-Infinity/blob/feature/fps-resolution-settings/src/electron/main.ts#L338-L345) and it's working. :)

Maybe we should set them as default ?

Elliesaur commented:
> I have tried your PR in my system (CPU AMD (R9 7950X3D ) and GPU AMD (RX 6900XT) on Bazzite OS but it doesn't work correctly at 1440P with the ultimate subscription. I have found [this one on Reddit,](https://www.reddit.com/r/LinuxOnAlly/comments/1kslnd1/get_better_decode_time_in_xcloud_on_bazzite/) and i added the argument `--enable-features=AcceleratedVideoDecodeLinuxZeroCopyGL,AcceleratedVideoDecodeLinuxGL,VaapiIgnoreDriverChecks` to the [/src/electron/main.ts](https://github.com/Elliesaur/GeForce-Infinity/blob/feature/fps-resolution-settings/src/electron/main.ts#L338-L345) and it's working. :)
> 
> Maybe we should set them as default ?

Can you tell me a bit more about how it doesn't work correctly? I'm interested to hear! I'm sure I can add those arguments although I'd need to research them to check they're not going to inadvertently impact other things.

t0msk commented:
it looks cool, thank you @Elliesaur you saved us lot of time, but I have several questions:

Did you try WarThunder? because in WT I cannot get more than 60 FPS (vsync is off)
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/db25e24e-a109-49ed-9ed6-9df3222ca1d7" />

Maybe it will enable HDR too? but I cannot test it, because I dont have HDR monitor

MrBretze commented:
> it looks cool, thank you @Elliesaur you saved us lot of time, but I have several questions:
> 
> Did you try WarThunder? because in WT I cannot get more than 60 FPS (vsync is off) <img alt="image" width="1920" height="1080" src="https://private-user-images.githubusercontent.com/7276200/477897782-db25e24e-a109-49ed-9ed6-9df3222ca1d7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxNTk1MzQsIm5iZiI6MTc1NTE1OTIzNCwicGF0aCI6Ii83Mjc2MjAwLzQ3Nzg5Nzc4Mi1kYjI1ZTI0ZS1hMTA5LTQ5ZWQtOWVkNi05ZGYzMjIyY2ExZDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MTRUMDgxMzU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTUzMTk0NDBlYTVlMmQzMzY5OThlN2EzZjk1ODZhZjg3M2UzNjZkM2Q4M2E4YzYxYzdmYjk4YTY2YmIwM2QxMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.qAP1SS1RlinGJ01baUzI3y5DeHvQTJrXkHBaADcLE5M">
> 
> Maybe it will enable HDR too? but I cannot test it, because I dont have HDR monitor

Probably a video decode issue, no?
In a Chrome browser, what is inside "Video Acceleration Information" in your chrome://gpu 

Can you try this argument ? `--enable-features=AcceleratedVideoDecodeLinuxZeroCopyGL,AcceleratedVideoDecodeLinuxGL,VaapiIgnoreDriverChecks`

t0msk commented:
Tried Fortnite and same issue

<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/a323eadc-5217-434d-81cd-e06045224e03" />

t0msk commented:
@Elliesaur @MrBretze my bad I have Performance tier and I thought that Performance is 120 FPS, so I cannot test FPS because I dont have Ultimate tier, can someone confirm that Fornite or WT (free games, dont have to buy it) works on 120 FPS?

So I can only test resolution and it works.

t0msk commented:
btw what about adding 30 FPS option? because i think there might be folks, who have low bandwidth internet and play low-FPS intense games like strategies, so it might save some bandwidth for them

Elliesaur commented:
> btw what about adding 30 FPS option? because i think there might be folks, who have low bandwidth internet and play low-FPS intense games like strategies, so it might save some bandwidth for them

You can do 30 fps, but it will set the resolution forcibly to 1366 x 720. I have added options for 30 FPS and 1366 x 720 resolution.

<img width="1915" height="1079" alt="image" src="https://github.com/user-attachments/assets/142a696b-dd24-42a8-b633-9b0f34f77a44" />

t0msk commented:
cool, so im fine with PR, just @MrBretze what is your issue on Bazzite?
